E3_Distill_ESRGAN/libs/train.py

```diff
@@ -1 +1,230 @@
+""" Trainer class to train student network to compress ESRGAN """
+
+from __future__ import absoulte_import
+from __future__ import division
+from __future__ import print_function
+
+import os
+
+from absl import logging
+from lib import dataset
+from libs import settings
+from libs import utils
 import tensorflow as tf
+
+
+class Trainer(object):
+  """Trainer Class for Knowledge Distillation of ESRGAN"""
+  def __init__(
+      self,
+      teacher,
+      discriminator,
+      summary_writer,
+      data_dir="",
+      manual=False,
+      model_dir=""):
+    """
+      Args:
+        teacher: Keras Model of pre-trained teacher generator.
+                 (Generator of ESRGAN)
+        discriminator: Keras Model of pre-trained teacher discriminator.
+                       (Discriminator of ESRGAN)
+        summary_writer: tf.summary.SummaryWriter object for writing
+                         summary for Tensorboard.
+        data_dir: Location of the stored dataset.
+        manual: Indicate if data_parameter contains Raw Files or TFRecords.
```

Comment:
nit: Keep 'similar'-types arguments together. So group together `data_dir` and `model_dir`. Also, `manual` doesn't seem the best way to describe what that parameter indicates. Maybe just `raw_data`. And also `data_parameter` -> `data_dir`?

Reply:
Any suggestions how to indicate if it is `data_dir` or `manual_dir`, without using `manual`?

Reply:
Wouldn't it be either of the two? So you can check which is not-empty?

Reply:
okay! Cool. :)

Reply:
PS: manual_dir and data_dir are parameters of `tfds.load(...)`.